example/glue/ConfigurationDelegate.hpp

```diff
@@ -0,0 +1,146 @@
+/*******************************************************************************
+ *
+ * (c) Copyright IBM Corp. 1991, 2016, 2017
+ *
+ *  This program and the accompanying materials are made available
+ *  under the terms of the Eclipse Public License v1.0 and
+ *  Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#ifndef CONFIGURATIONDELEGATE_HPP_
+#define CONFIGURATIONDELEGATE_HPP_
+
+#include "omrgcconsts.h"
+#include "sizeclasses.h"
+
+#include "EnvironmentBase.hpp"
+#include "EnvironmentLanguageInterfaceImpl.hpp"
+#include "GCExtensionsBase.hpp"
+#include "Heap.hpp"
+
+/**
+ * Client language delegate for GC configuration must be implemented by all GC clients.
+ */
+class MM_ConfigurationDelegate
+{
+/*
+ * Member data and types
+ */
+private:
+	const MM_GCPolicy _gcPolicy;
+
+protected:
+public:
+
+/*
+ * Member functions
+ */
+private:
+protected:
+
+public:
+	/**
+	 * Initialize the delegate.
+	 * @param env The calling thread environment
+	 * @param writeBarrierType The write barrier type selected for the GC configuration
+	 * @param allocationType The allocation type selected for the GC configuration
+	 * @return false if initialization fails
+	 */
+	bool
+	initialize(MM_EnvironmentBase* env, MM_GCWriteBarrierType writeBarrierType, MM_GCAllocationType allocationType)
+	{
+		return true;
+	}
+
+	/**
+	 * Tear down the delegate, releasing any resources held.
+	 * @param env The calling thread environment
+	 */
+	void
+	tearDown(MM_EnvironmentBase* env)
+	{
+		return;
+	}
+
+	/**
+	 * This method is called just after the heap has been initialized. Delegate may perform
+	 * any additional heap-related initialization at this point.
+	 * @param env The calling thread environment
+	 * @return false if heap initialization fails
+	 */
+	bool
+	initializeHeap(MM_EnvironmentBase* env)
```

Comment:
I agree. This and initializeEnvironment() are both informational (hooks) called from OMR to permit language to respond to GC-related events. Using your suggested name pattern for both of these and will carry this convention forward into ELI and GCLI component delgates.